src/lib/lspace/layout/lalloc.rs

```diff
@@ -0,0 +1,805 @@
+                    // More space than natural size
+                    if child_req.flex().stretch() > 0.0 {
+                        // Stretch to take up additional space
+                        self.update_ref(region_pos, region_size, region_size, alloc_before);
+                    } else {
+                        // No stretch; don't take up additional space
+                        self.update_ref(ref_offset + region_pos, natural_size, natural_size,
+                                        alloc_before - ref_offset);
+                    }
+                } else {
+                    // region_size is between minimum and natural end
+                    let alloc_size = region_size - ref_offset;
+                    self.update_ref(ref_offset + region_pos, alloc_size, alloc_size,
+                                    alloc_before - ref_offset);
+                }
+            },
+
+            // No reference point required
+            None => {
+                let natural_size = child_req.size().size();
+                let minimum_size = natural_size - child_req.flex().shrink();
+
+                if region_size < minimum_size * ONE_MINUS_EPSILON {
+                    // Insufficient space; allocate the available space, while giving
+                    // `minimum_size` as the actual size for bounding box and rendering purposes
+                    self.update(region_pos, region_size, minimum_size);
+                } else if region_size < natural_size * ONE_PLUS_EPSILON ||
+                          child_req.flex().stretch() > 0.0 {
+                    // Either:
+                    // - region_size is between minimum and natural size
+                    // or
+                    // - region_size is > natural size and the box can stretch to use the
+                    // additional space
+                    //
+                    // Allocate all the available space
+                    self.update(region_pos, region_size, region_size);
+                } else {
+                    // region_size is > natural and no stretch; don't take up additional space
+                    self.update(region_pos, natural_size, natural_size);
+                }
+            }
+        }
+    }
+
+
+    fn _alloc_children_linear_fixed(n: usize,
+                                    child_reqs: &[&LReq], child_allocs: &mut [&mut LAlloc],
+                                    start_pos: f64, space_between: f64) -> f64 {
+        let mut pos = start_pos;
+        for i in 0..n {
+            let creq = child_reqs[i];
+            let csz = creq.size().size();
+            child_allocs[i].alloc_from_region(creq, pos, csz, creq.size().before_ref_opt());
+            pos = pos + csz + space_between;
+        }
+        return pos;
+    }
+
+    fn _alloc_children_linear_flex(n: usize,
+                                   child_reqs: &[&LReq], child_allocs: &mut [&mut LAlloc],
```

Comment:
Our of curiosity, why have these as (in effect) static methods on the struct? They don't seem to be directly related to it (though they are tangentially related).